codec_test.go

```diff
@@ -90,7 +91,7 @@ type fromStringTest struct {
 }
 
 // Run runs the FromString test in a subtest of t, named by fst.variant.
-func (fst fromStringTest) Run(t *testing.T) {
+func (fst fromStringTest) TestFromString(t *testing.T) {
```

Comment:
Why was this method renamed?  The convention with these tests is that each one has a `Run()` method that does the needful.

It would be preferable to add a 2nd `t.Run()` than rename the method and add a separate `TestUnmarshalText()` method with a single `t.Run()`.

Reply:
I don't have a strong opinion here, but the method was renamed because I added `TestUnmarshalText` to the `fromStringTest` type (and IMHO neither method should exist and the logic should actually live in the top-level test functions).

Since we can't have two `Run()` methods, we are testing two unique code paths, and duplicating this type and it's test cases (just to have a `Run()` method) seems like an undue maintenance burden. I thought it would be better to add the `TestUnmarshalText` method and rename `Run` to `TestFromString`.

> It would be preferable to add a 2nd t.Run() than rename the method and add a separate TestUnmarshalText() method with a single t.Run().

IMHO, invoking the `UnmarshalText` tests separately from the top-level `TestUnmarshalText` function makes for cleaner test output (e.g. `TestUnmarshalText/Valid/Canonical`. If we had two `Run` methods then the output (depending on how its implemented) would look something like:
```
TestFromString
TestFromString/Valid
TestFromString/Valid/Canonical
...
# UnmarshalText (test names are repeated)
TestFromString
TestFromString/Valid
TestFromString/Valid/Canonical
```

Or if we name the subtest "UnmarshalText" it'll look like:
```
TestFromString
TestFromString/Valid
TestFromString/Valid/Canonical
...
TestFromString/UnmarshalText
TestFromString/UnmarshalText/Valid
TestFromString/UnmarshalTextValid/Canonical
```

Reply:
fwiw I don't have a strong opinion either.  My comment was mostly about sticking to the existing conventions, which pre-dated me.

Reply:
You've convinced me.  After another pass over the code I agree with the change, if for slightly different reasons (`fromStringTest` is the only type here with a `Run()` method and it's there to wrap executing the batch of tests.  we're doing 2 batches so 2 names does make sense).